tools/ReleaseProcess/UploadArtifactsToGitHub.ps1

```diff
@@ -0,0 +1,459 @@
+## Copyright (c) Microsoft. All rights reserved.
+## Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+    .SYNOPSIS
+        This function downloads and prepares the Iot Edge Daemon artifacts from DevOps pipeline for a release. 
+        The final artifacts are collected under "<WorkDir>/output" path.
+    .DESCRIPTION
+        1) Download edgelet artifacts from a given BuildId from a DevOps pipeline
+        2) Extract the *.zip artifacts into their own directories
+        3) Traverse the directories and rename a relavant artifact into the following format: 
+              .DEB _________________ {component}_{version}_{os}_{arch}.deb 
+              .RPM _________________ {component}-{version}.{os}.{arch}.rpm  (No name change)
+        4) Move the renamed artifacts to a staging directory at "<WorkDir>/output"
+    .INPUTS 
+        $DevOpsPat ____________ (String) Environment variable contains DevOps Personal Access Token (with Build:Read privilege) in UTF8 encoding
+    .PARAMETER BuildId
+        DevOps build ID for the downloading artifacts
+    .PARAMETER WorkDir
+        Absolute path of the working directory
+#>
+function Prepare-DevOps-Artifacts
+{
+    [CmdletBinding()]
+    param (
+        <# 
+        DevOps build ID for the downloading artifacts
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $BuildId,
+
+        <# 
+        Absolute path of the working directory
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $WorkDir
+    )
+
+    if ([string]::IsNullOrEmpty($DevOpsPat))
```

Comment:
It seems like this shouldn't work because `$DevOpsPat` isn't defined in this script. If it's a variable in the environment, doesn't it have to be `$env:DevOpsPat`?

Reply:
I created a dummy function to test this 
``` pwsh
function Test
{
    if ([string]::IsNullOrEmpty($DevOpsPat))
    {
        echo 'Environment variable ($DevOpsPat) is required.'
        return 1;
    }
    else
    {
        $pat = $DevOpsPat;
    }
}
```
It seems to work.

Reply:
Ah...okay. Since the caller has to dot-source the script into their current PowerShell session and then call the function(s) directly, any "local" session variables will be available to the functions. So you're right. But `$DevOpsPat` is not an environment variable, so calling it that might lead to confusion.

Reply:
I think if you just remove the word "environment" from the script, you're good to go.